### PR TITLE
CB-6264 Add Spot Percentage to DistroX request

### PR DESCRIPTION
**Pull request by bergerdenes**

### CODE REVIEW

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/view/AwsInstanceView.java [hunk 1]
```diff
@@ -74,15 +74,7 @@ public String getKmsKey() {
...
-        if (sv instanceof Double) {
-            return (Double) sv;
-        } else if (sv instanceof Integer) {
-            return ((Number) sv).doubleValue();
-        } else if (sv instanceof String) {
-            return Double.valueOf((String) sv);
-        }
-        return null;
+    public Integer getSpotPercentage() {
+        return instanceTemplate.getParameter("spotPercentage", Integer.class);
```
Comment (doktoric) [inline cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/view/AwsInstanceView.java:78]: isn't it just percentage?

File: cloud-aws/src/main/resources/templates/aws-cf-stack.ftl [hunk 2]
```diff
@@ -299,7 +299,7 @@
           },
           "InstancesDistribution": {
             "OnDemandBaseCapacity": 0,
-            "OnDemandPercentageAboveBaseCapacity": 100,
+            "OnDemandPercentageAboveBaseCapacity": ${100 - (group.spotPercentage!0)},
```
Comment (doktoric) [inline cloud-aws/src/main/resources/templates/aws-cf-stack.ftl:302]: I think we should use here the OnDemandPercentage and calculate the value in the java class

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/view/AwsInstanceView.java (hunk 1 above)
Comment (bergerdenes) [reply cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/view/AwsInstanceView.java:78]: on api level it is, but as internal parameter accompanying the instanceTemplate sitting in a map it is more clear, I guess

File: cloud-aws/src/main/resources/templates/aws-cf-stack.ftl (hunk 2 above)
Comment (bergerdenes) [reply cloud-aws/src/main/resources/templates/aws-cf-stack.ftl:302]: well, yes, let's do it that way

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/view/AwsInstanceView.java (hunk 1 above)
Comment (doktoric) [reply cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/view/AwsInstanceView.java:78]: ok

**Pull request merged**